# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/mediasink/AudioSinkWrapper.cpp
# Commit: 4d427e0ca400
# Full Hash: 4d427e0ca4008a8e00f40b05fcc5050035921ddc
# Author: Paul Adenot <paul@paul.cx>
# Date: 2022-04-20 21:53:00
# Regressor Bug: 1743834
# File Overlap Count: 1
# Description:
#   Bug 1743834 - Detect when switching clock provider in AudioSinkWrapper. r=alwu,media-playback-reviewers
#   
#   This will allow detecting the first time the AudioSinkWrapper starts using the
#   clock from an audio stream after having been muted (=using the system clock) for
#   some time, to correct the timing.
# ==============================================================================

diff -r 3000680ad5f4 -r 4d427e0ca400 dom/media/mediasink/AudioSinkWrapper.cpp
--- a/dom/media/mediasink/AudioSinkWrapper.cpp	Wed Apr 20 12:07:10 2022 +0000
+++ b/dom/media/mediasink/AudioSinkWrapper.cpp	Wed Apr 20 12:07:10 2022 +0000
@@ -81,6 +81,7 @@
     // Rely on the audio sink to report playback position when it is not ended.
     pos = mAudioSink->GetPosition();
     LOGV("%p: Getting position from the Audio Sink %lf", this, pos.ToSeconds());
+    mLastClockSource = ClockSource::AudioStream;
   } else if (!mPlayStartTime.IsNull()) {
     // Calculate playback position using system clock if we are still playing,
     // but not rendering the audio, because this audio sink is muted.
@@ -100,10 +101,12 @@
         mEndedPromiseHolder.Resolve(true, __func__);
       }
     }
+    mLastClockSource = ClockSource::SystemClock;
   } else {
     // Return how long we've played if we are not playing.
     pos = mPlayDuration;
     LOGV("%p: Getting static position, not playing %lf", this, pos.ToSeconds());
+    mLastClockSource = ClockSource::Paused;
   }
 
   if (aTimeStamp) {